### ISSUE TITLE
BIDS cli does not follow Symlink directories inside the dataset

### ISSUE THREAD

**Issue by naveau** [bids-standard/legacy-validator#575]
I came to this error trying to validate a "sub"-dataset built by linking only some session directories from a main dataset in each subject directories.

Here is the tree of the "sub"-dataset built from `ds001`:
```
[...]
|-- sub-active36                                                                                                                                                                                               
|   |-- ses-posttraining -> ../../ds001/sub-active36/ses-posttraining                                                                                                                                          
|   `-- ses-pretraining -> ../../ds001/sub-active36/ses-pretraining                                                                                                                                            
|-- sub-active37                                                                                                                                                                                               
|   |-- ses-posttraining -> ../../ds001/sub-active37/ses-posttraining                                                                                                                                          
|   `-- ses-pretraining -> ../../ds001/sub-active37/ses-pretraining                                                                                                                                            
|-- sub-inhibition01                                                                                                                                                                                           
|   |-- ses-posttraining -> ../../ds001/sub-inhibition01/ses-posttraining                                                                                                                                      
|   `-- ses-pretraining -> ../../ds001/sub-inhibition01/ses-pretraining                                                                                                                                        
|-- sub-inhibition02                                                                                                                                                                                           
|   |-- ses-posttraining -> ../../ds001/sub-inhibition02/ses-posttraining                                                                                                                                      
|   `-- ses-pretraining -> ../../ds001/sub-inhibition02/ses-pretraining
[...]
```

Everything's working fine when I ran the validator in the browser but the cli version failed at the `Quick Test` :

```
[ERR]  The directory dsproject01/ failed an initial Quick Test.
```

I assumed that it was due to the symbolic link and attempt a monkey patch that work in my situation : https://github.com/naveau/bids-validator/commit/ce9b660abe5587f021ff015f79d80bbeab2d4c85

Basically, I test if the files read by `fs.readDirSync` are Symbolic Links pointing to a directory.

I don't know if this is a case you want to cover in the code and if there is a better way to handle it. I can create a WIP pull request if you provide me some hints.

**Comment by chrisgorgo** [comment]
This sounds like a regression. Thanks for reporting. We should definitely cover such cases. Would you mind opening a Pull Request with your fix?
